lua/starfall/libs_sv/nextbot.lua

```diff
@@ -0,0 +1,598 @@
+registerprivilege("nextbot.setStepHeight", "Set nextbot step height", "Allows the user to set the nextbot's step height", {entities = {}})
+registerprivilege("nextbot.jumpAcrossGap", "Nextbot jump across gap", "Allows the user to make a nextbot jump across a gap.", {entities = {}})
+registerprivilege("nextbot.setClimbAllowed", "Nextbot allow climb", "Allows the user to set whether the nextbot can climb nav ladders.", {entities = {}})
+registerprivilege("nextbot.setAvoidAllowed", "Nextbot allow avoid", "Allows the user to set whether the nextbot can try to avoid obstacles.", {entities = {}})
+registerprivilege("nextbot.setJumpGapsAllowed", "Nextbot allow jump gaps", "Allows the user to set whether the nextbot can jump gaps.", {entities = {}})
+
+local nbCount = SF.LimitObject("nextbots", "nextbots", 30, "The number of props allowed to spawn via Starfall")
+
+return function(instance)
+local checkpermission = instance.player ~= SF.Superuser and SF.Permissions.check or function() end
+
+local nextbots = {}
+
+local nextbot_library, nb_meta, nb_methods = instance.Libraries.nextbot, instance.Types.NextBot, instance.Types.NextBot.Methods
+local vec_meta, vwrap, vunwrap = instance.Types.Vector, instance.Types.Vector.Wrap, instance.Types.Vector.Unwrap
+local navarea_methods, navarea_meta, navwrap, navunwrap = instance.Types.NavArea.Methods, instance.Types.NavArea, instance.Types.NavArea.Wrap, instance.Types.NavArea.Unwrap
+local nbwrap, nbunwrap = instance.Types.NextBot.Wrap, instance.Types.NextBot.Unwrap
+
+local function nextbotOnDestroy(ent)
+	local ply = instance.player
+	nbCount:free(ply, 1)
+	nextbots[ent] = nil
+end
+
+local function register(ent)
+	ent:CallOnRemove("starfall_nextbot_delete", nextbotOnDestroy)
+	nbCount:free(instance.player, -1)
+	nextbots[ent] = true
+end
+
+instance:AddHook("deinitialize", function()
+	if #nextbots > 0 then
+		for nextbot, _ in pairs(nextbots) do
+			-- For some reason there's numbers being added to the nextbots table when they're created so I have to do this or it errors. Why does it add numbers to the table? I wish I knew.
+			if IsEntity(nextbot) then
+				nextbot:Remove()
+			end
+		end
+	end
+end)
+
+function nb_meta:__tostring()
+	return "NextBot"
+end
+
+--- Creates a customizable NextBot
+-- @server
+-- @param Vector spawnpos The position the nextbot will be spawned at.
+-- @param string? model The model the nextbot will use. If nil, uses Kleiner.
+-- @return NextBot The nextbot.
+function nextbot_library.create(pos, mdl)
+
+	checkpermission(instance, nil, "nextbot.create")
+	checkluatype(mdl, TYPE_STRING)
+	
+	local ply = instance.player
+	nbCount:checkuse(ply, 1)
+
+	local nb = ents.Create("starfall_cnextbot")
+	register(nb, instance)
+	nb:SetPos(vunwrap(pos))
+	nb:SetNBModel(mdl or "models/kleiner.mdl")
+	nb.chip = instance.entity
+	nb:Spawn()
+	table.insert(nextbots, nb)
```

Comment:
This is why numbers are getting added to the table. This should be `nextbots[nb] = true`